src/tracker.js

```diff
@@ -0,0 +1,18 @@
+const createTracker = page => {
```

Comment:
This could do with a doc string. I don't know what the right JSDoc notation should be but we ought to explain its purpose in life. 

*"A function that sets up request events on a Puppeteer.Page instance so that we can keep track of which URLs have started being request and which have finished. Ultimately its purpose is it be able to, at any point (e.g. an unexpected navigation error), be able to say which URLs have *not* finished. Having this list of URLs can be helpful when debugging why you get unpredictable timeouts."*